lstchain/io/containers.py

```diff
@@ -1,154 +1,268 @@
-        except:
-            print("event information not filled")
-
-    def get_features(self, features_names):
-        return np.array([self[k].value if isinstance(self[k], Quantity) else self[k]
-                         for k in features_names])
-
-    def set_mc_core_distance(self, event, telescope_id):
-        tel_pos = event.inst.subarray.positions[telescope_id]
-        distance = np.sqrt((event.mc.core_x - tel_pos[0]) ** 2 + (event.mc.core_y - tel_pos[1]) ** 2)
-        self.mc_core_distance = distance
-
-    def set_disp(self, source_pos, hillas):
-        disp = utils.disp_parameters(hillas, source_pos[0], source_pos[1])
-        self.disp_norm = disp.norm
-        self.disp_dx = disp.dx
-        self.disp_dy = disp.dy
-        self.disp_angle = disp.angle
-        self.disp_sign = disp.sign
-        self.disp_miss = disp.miss
-
-    def set_timing_features(self, geom, image, peakpos, hillas):
-        try:
-            peak_time = Quantity(peakpos) * u.Unit("ns")
-            timepars = time.timing_parameters(geom, image, peak_time, hillas)
-            self.time_gradient = timepars['slope'].value
-            self.intercept = timepars['intercept']
-        except:
-            pass
-
-    def set_source_camera_position(self, event, telescope_id):
-        # sourcepos = utils.cal_cam_source_pos(mc_alt, mc_az,
-        #                                      mc_alt_tel, mc_az_tel,
-        #                                      focal_length)
-        # self.src_x = sourcepos[0]
-        # self.src_y = sourcepos[1]
-        tel = event.inst.subarray.tel[telescope_id]
-        source_pos = utils.get_event_pos_in_camera(event, tel)
-        self.src_x = source_pos[0]
-        self.src_y = source_pos[1]
-
-
-class DispContainer(Container):
-    """
-    Disp vector container
-    """
-    dx = Field(nan, 'x coordinate of the disp_norm vector')
-    dy = Field(nan, 'y coordinate of the disp_norm vector')
-
-    angle = Field(nan, 'Angle between the X axis and the disp_norm vector')
-    norm = Field(nan, 'Norm of the disp_norm vector')
-    sign = Field(nan, 'Sign of the disp_norm')
-    miss = Field(nan, 'miss parameter norm')
+from astropy import units as u
+#from astropy.time import Time
+#from numpy import nan
+#import numpy as np
+
+from ctapipe.core import Container, Field, Map
+from ctapipe.io.containers import *
+
+__all__ = [
+    'FlatFieldCameraContainer',
+    'FlatFieldContainer',
+    'PedestalCameraContainer',
+    'PedestalContainer',
+    'PixelStatusCameraContainer',
+    'PixelStatusContainer',
+    'LSTCameraContainer',
+    'MonitoringDataContainer',
+    'LSTDataContainer'
+]
+
+
+class R0CameraContainer(Container):
+    """
+    Storage of raw data from a single telescope
+    """
+    trigger_time = Field(None, "Telescope trigger time, start of waveform "
+                               "readout, None for MCs")
+    trigger_type = Field(0o0, "camera's event trigger type if applicable")
```

Comment:
Out of curiosity, why `0o0` ?

Reply:
:-) no idea, I wondered also the meaning of this.

Anyway, I just changed the code in order not to have the R0CameraContainer in lstchain